### PR TITLE
Notification Comment Details: enable feature for release

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref: #17790

This enables the new Comment details view for Comment Notifications.

To test:

TLDR - verify everything works as expected.

- Go to Notifications > comment notification.
- Verify the new details view is displayed.
- Verify Spam, Trash, and deleted comments are removed from the notifications list.
- Verify the ⬆️ and ⬇️ nav bar buttons display the previous and next notification.
- Verify Edit, Reply, and Like work as expected.
- Verify the `You replied...` message appears if you've replied to a comment.

If you do not have permission to the site the notification is for, verify:
- The comment content appears.
- The moderation bar does not.
- `Edit` nav bar button does not.
- `Email address` and `IP address` do not.

| all_the_things | some_things |
|--------|-------|

## Regression Notes
1. Potential unintended areas of impact
Should be done.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested Notification Comments and My Site Comments to be sure both work as expected.

3. What automated tests I added (or what prevented me from doing so)
Updated existing WP tests, new tests added to WPKit.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by dvdchr — APPROVED** [review]
🎉
